client/src/client.ts

```diff
@@ -1671,7 +1671,22 @@ class CodeActionFeature extends TextDocumentFeature<TextDocumentRegistrationOpti
 	}
 
 	public fillClientCapabilities(capabilites: ClientCapabilities): void {
-		ensure(ensure(capabilites, 'textDocument')!, 'codeAction')!.dynamicRegistration = true;
+		const cap = ensure(ensure(capabilites, 'textDocument')!, 'codeAction')!;
+		cap.dynamicRegistration = true;
+		cap.codeActionLiteralSupport = {
+			codeActionKind: {
+				valueSet: [
+					VCodeActionKind.Empty.value!,
+					VCodeActionKind.QuickFix.value!,
+					VCodeActionKind.Refactor.value!,
+					VCodeActionKind.RefactorExtract.value!,
+					VCodeActionKind.RefactorInline.value!,
+					VCodeActionKind.RefactorRewrite.value!,
+					VCodeActionKind.Source.value!,
+					VCodeActionKind.SourceOrganizeImports.value!
```

Comment:
Wouldn't it make more sense to use `CodeActionKind` from the language server protocol instead of using the VS Code one? That sounds a lot safer and is more agnostic since then we're only using stuff that the protocol explicitly declares.

Reply:
Agree. Will change it.